src/js/components/Diagram/Diagram.js

```diff
@@ -222,20 +273,49 @@ const Diagram = forwardRef(({ connections, ...rest }, ref) => {
             colorName = colors[index % colors.length];
           }
 
+          let arrowMarker = null;
+
+          if (typeof endpoint === 'object' && endpoint?.from === 'arrow') {
+            arrowMarker = openArrow(
+              normalizeColor(colorName, theme),
+              index,
+              'openArrowStart',
+              'auto-start-reverse',
+            );
+          } else if (typeof endpoint === 'object' && endpoint?.to === 'arrow') {
+            arrowMarker = openArrow(normalizeColor(colorName, theme), index);
+          } else if (typeof endpoint === 'string' && endpoint === 'arrow') {
+            arrowMarker = (
+              <>
+                {openArrow(
+                  normalizeColor(colorName, theme),
+                  index,
+                  'openArrowStart',
+                  'auto-start-reverse',
+                )}
+                {openArrow(normalizeColor(colorName, theme), index)}
+              </>
+            );
+          }
+
           path = (
-            <path
-              // eslint-disable-next-line react/no-array-index-key
-              key={index}
-              // eslint-disable-next-line react/no-unknown-property
-              animation={animation}
-              {...cleanedRest}
-              stroke={normalizeColor(colorName, theme)}
-              strokeWidth={strokeWidth}
-              strokeLinecap={round ? 'round' : 'butt'}
-              strokeLinejoin={round ? 'round' : 'miter'}
-              fill="none"
-              d={d}
-            />
+            // eslint-disable-next-line react/no-array-index-key
+            <Fragment key={index}>
+              <path
+                // eslint-disable-next-line react/no-unknown-property
+                animation={animation}
+                {...cleanedRest}
+                stroke={normalizeColor(colorName, theme)}
+                strokeWidth={strokeWidth}
+                strokeLinecap={round ? 'round' : 'butt'}
+                strokeLinejoin={round ? 'round' : 'miter'}
+                fill="none"
+                d={d}
+                markerStart={`url("#openArrowStart-${index}")`}
+                markerEnd={`url("#openArrowEnd-${index}")`}
+              />
+              {endpoint && arrowMarker && <defs>{arrowMarker}</defs>}
```

Comment:
Why do we wrap in `defs`?

Reply:
The main use case for `defs`  was to holds a set of reusable definitions for the SVG markers. But, it isn't serving it's purpose right now, I will push some change in which we will using arrow based on it's open, close and color.